docs/Testing.md

```diff
@@ -1,13 +1,129 @@
 <h1 align="center">Fastify</h1>
 
 ## Testing
+
 Testing is one of the most important parts of developing an application. Fastify is very flexible when it comes to testing and is compatible with most testing frameworks (such as [Tap](https://www.npmjs.com/package/tap), which is used in the examples below).
 
-<a name="inject"></a>
-### Testing with http injection
+Let's `cd` into a fresh directory called 'testing-example' and type `npm init -y` in our terminal.
+
+run `npm install fastify && npm install tap pino-pretty --save-dev`
+
+### Separating concerns makes testing easy
+
+ First we're going to separate our application code from our server code:
+
+app.js
+
+```js
+'use strict'
+
+const Fastify = require('fastify')
+
+function build(opts={}) {
+  const app = Fastify(opts)
+  app.get('/', async function (request, reply) {
+    return { hello: 'world' }
+  })
+
+  return app
+}
+
+module.exports = build
+```
+
+server.js
+
+```js
+'use strict'
+
+const server = require('./app')({
+    logger: {
+      level: 'info',
+      prettyPrint: true
+    }
+  })
```

Comment:
The indentation is weird here. It should be shifted to the left to match the following `server.listen` block.